addons/portal/controllers/mail.py

```diff
@@ -112,6 +112,19 @@ def _portal_post_has_content(self, res_model, res_id, message, attachment_ids=No
         """ Tells if we can effectively post on the model based on content. """
         return bool(message) or bool(attachment_ids)
 
+    @http.route('/mail/avatar/mail.message/<int:res_id>/author_avatar/<int:width>x<int:height>', type='http', auth='public')
+    def portal_avatar(self, res_id=None, height=50, width=50, access_token=None):
+        """ Get the avatar image in the chatter of the portal """
+        record_sudo = request.env.ref('web.image_placeholder').sudo()
```

Comment:
I don't think this `sudo` here is needed
```suggestion
        record_sudo = request.env.ref('web.image_placeholder')
```